toolkit/tools/imagegen/configuration/configuration.go

```diff
@@ -92,13 +68,69 @@ type Config struct {
 	DefaultSystemConfig *SystemConfig // A system configuration with the "IsDefault" field set or the first system configuration if there is no explicit default.
 }
 
+// GetDiskPartByID returns the disk partition object with the desired ID, nil if no partition found
+func (c *Config) GetDiskPartByID(ID string) (disk *Partition) {
+	for i, d := range c.Disks {
+		for j, p := range d.Partitions {
+			if p.ID == ID {
+				return &c.Disks[i].Partitions[j]
+			}
+		}
+	}
+	return nil
+}
+
+// checkDeviceMapperFlags checks if Encryption and read-only roots have the required 'dmroot' flag.
+// They need the root partition to have a specific flag so we can find the partition and handle it
+// before we mount it.
+func checkDeviceMapperFlags(config *Config) (err error) {
+	for _, sysConfig := range config.SystemConfigs {
+		var dmRoot *Partition = nil
+		if sysConfig.Encryption.Enable {
+			rootPartSetting := sysConfig.GetRootPartitionSetting()
+			if rootPartSetting == nil {
+				return fmt.Errorf("can't find a root ('/') [PartitionSetting] to work with either [ReadOnlyVerityRoot] or [Encryption]")
```

Comment:
>ReadOnlyVerityRoot [](start = 87, length = 18)

Since you're not checking for `sysConfig.ReadOnlyVerityRoot.Enable` anymore, shouldn't the error message change as well?

Reply:
Just moved to later, I wanted to minimize the number of partial line changes I needed when I pulled those bits out to make my life easier in the future.

---
In reply to: [549906163](https://github.com/microsoft/CBL-Mariner/pull/479#discussion_r549906163) [](ancestors = 549906163)